src/lava/magma/compiler/compiler.py

```diff
@@ -196,6 +196,34 @@ def _select_proc_models(
         """Selects a ProcessModel from list of provided models given RunCfg."""
         return run_cfg.select(proc, models)
 
+    @staticmethod
+    def _propagate_var_ports(proc: AbstractProcess):
+        """Checks if the process has VarPorts configured and if they reference
+        an aliased Var. If this is the case an implicit VarPort is created in
+        the sub process and the VarPort is connected to the new implicit
+        VarPort.
+        This is needed since a RefPort or VarPort of a sub process cannot be
+         directly targeted from processes of a different parent process."""
+
+        for vp in proc.var_ports:
+            v = vp.var.aliased_var
+            if v is not None:
+                sub_proc = v.process
+                # Create an implicit Var port in the sub process
+                new_vp = ImplicitVarPort(v)
+                # Propagate name and parent process of Var to VarPort
+                new_vp.name = "_" + v.name + "_implicit_port"
```

Comment:
I think we did this already somewhere else. Could be factored out into a shared function.

Reply:
correct